prusti-viper/src/encoder/name_interner.rs

```diff
@@ -4,60 +4,130 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+use prusti_common::config;
+//use vir::common::identifier::NameHash;
 use rustc_hash::{FxHashMap, FxHashSet};
+use rustc_middle::ty::TyCtxt;
+use rustc_span::def_id::DefId;
+use std::fmt::{Display, Formatter, Result};
 use log::debug;
 
+// Tracks if crate-unqiue and module-unique names are needed to ensure uniqueness
+#[derive(Debug)]
+struct ShortName {
+    prefix: String,
+    project_unique: (bool, String),
+    crate_unique: (bool, String),
+    mod_unique: (bool, String),
+    short_name: String,
```

Comment:
`ShortName { path: Vec<String>, suffix_len: usize }` with `1 <= suffix_len <= path.len()` would be more generic and might even require less code.